extensions/wikia/WallNotifications/WallNotificationEntity.class.php

```diff
@@ -78,174 +127,202 @@ public function getUniqueId() {
 		} else {
-			$this->data->reason = '';
+			$acParent->load();
+			$title = $acParent->getTitle();
+			$this->parentTitleDbKey = $title->getDBkey();
+			$this->threadTitleFull = $acParent->getMetaTitle();
+
+			$this->setMessageParentUserData( $data, $acParent );
+			$data->parent_id = $acParent->getId();
+			$data->thread_title = $acParent->getMetaTitle();
 		}
+	}
 
-		if ( !empty( $acParent ) ) {
-			$acParent->load();
-			$parentUser = $acParent->getUser();
-
-			if ( $parentUser instanceof User ) {
-				$this->data->parent_username = $parentUser->getName();
-				if ( $parentUser->getId() > 0 ) {
-					$this->data->parent_displayname = $this->data->parent_username;
-				} else {
-					$this->data->parent_displayname = wfMessage( 'oasis-anon-user' )->text();
-				}
-				$this->data->parent_user_id = $acParent->getUser()->getId();
+	private function setMessageParentUserData( stdClass $data, WallMessage $parent ) {
+		$parentUser = $parent->getUser();
+
+		if ( $parentUser instanceof User ) {
+			$data->parent_username = $parentUser->getName();
+			$data->parent_user_id = $parentUser->getId();
+
+			if ( $data->parent_user_id > 0 ) {
+				$data->parent_displayname = $data->parent_username;
 			} else {
+				$data->parent_displayname = wfMessage( 'oasis-anon-user' )->text();
+			}
+		} else {
 			/* parent was deleted and somehow reply stays in the system
 			 * the only way I've reproduced it was: I deleted a thread
 			 * then I went to Special:Log/delete and restored only its reply
 			 * an edge case but it needs to be handled
 			 * --nAndy
 			 */
-				$this->data->parent_username = $this->data->parent_displayname = wfMessage( 'oasis-anon-user' )->text();
-				$this->data->parent_user_id = 0;
-			}
-			$title = $acParent->getTitle();
-			$this->data_noncached->thread_title_full = $acParent->getMetaTitle();
-			$this->data->thread_title = $acParent->getMetaTitle();
-			$this->data_noncached->parent_title_dbkey = $title->getDBkey();
-			$this->data->parent_id = $acParent->getId();
-			$this->data->url = $wm->getMessagePageUrl();
-
-		} else {
-			$this->data->url = $wm->getMessagePageUrl();
-			$this->data->parent_username = $walluser->getName();
-			$this->data_noncached->thread_title_full = $wm->getMetaTitle();
-			$this->data->thread_title = $wm->getMetaTitle();
-		}
-
-		return true;
-	}
-
-	protected function buildId( $wikiId, $RCid ) {
-
-	}
-
-	function recreateFromDB() {
-		$explodedId = explode( '_', $this->id );
-		$RevId = $explodedId[0];
-		$wikiId = $explodedId[1];
-
-		$rev = Revision::newFromId( $RevId );
-		if ( empty( $rev ) ) {
-			/* also cache failures not to make expensive database queries
-			 * again and again for the same Entity */
-			$this->save();
-			return;
+			$data->parent_username = wfMessage( 'oasis-anon-user' )->text();
+			$data->parent_displayname = $data->parent_username;
+			$data->parent_user_id = 0;
 		}
-
-		$this->loadDataFromRev( $rev, $wikiId );
-		$this->save();
 	}
 
```

Comment:
I thought ```save``` was misleading since most of the time this implies permanent storage.  Updated to declare that its actually saving to cache.